server.go

```diff
@@ -931,95 +932,20 @@ func newServer(ctx context.Context, cfg *Config, listenAddrs []net.Addr,
 	selfAddrs = append(selfAddrs, externalIPs...)
 
-	// We'll now reconstruct a node announcement based on our current
-	// configuration so we can send it out as a sort of heart beat within
-	// the network.
-	//
-	// We'll start by parsing the node color from configuration.
-	color, err := lncfg.ParseHexColor(cfg.Color)
-	if err != nil {
-		srvrLog.Errorf("unable to parse color: %v\n", err)
-		return nil, err
-	}
-
-	// If no alias is provided, default to first 10 characters of public
-	// key.
-	alias := cfg.Alias
-	if alias == "" {
-		alias = hex.EncodeToString(serializedPubKey[:10])
-	}
-	nodeAlias, err := lnwire.NewNodeAlias(alias)
-	if err != nil {
-		return nil, err
-	}
-
 	// TODO(elle): All previously persisted node announcement fields (ie,
 	//  not just LastUpdate) should be consulted here to ensure that we
 	//  aren't overwriting any fields that may have been set during the
 	//  last run of lnd.
-	nodeLastUpdate := time.Now()
-	srcNode, err := dbs.GraphDB.SourceNode(ctx)
-	switch {
-	// If we have a source node persisted in the DB already, then we just
-	// need to make sure that the new LastUpdate time is at least one
-	// second after the last update time.
-	case err == nil:
-		if srcNode.LastUpdate.Second() >= nodeLastUpdate.Second() {
-			nodeLastUpdate = srcNode.LastUpdate.Add(time.Second)
-		}
-
-	// If we don't have a source node persisted in the DB, then we'll
-	// create a new one with the current time as the LastUpdate.
-	case errors.Is(err, graphdb.ErrSourceNodeNotSet):
 
-	// If the above cases are not matched, then we have an unhandled non
-	// nil error.
-	default:
-		return nil, fmt.Errorf("unable to fetch source node: %w", err)
-	}
-
-	selfNode := &models.LightningNode{
-		HaveNodeAnnouncement: true,
-		LastUpdate:           nodeLastUpdate,
-		Addresses:            selfAddrs,
-		Alias:                nodeAlias.String(),
-		Features:             s.featureMgr.Get(feature.SetNodeAnn),
-		Color:                color,
-	}
-	copy(selfNode.PubKeyBytes[:], nodeKeyDesc.PubKey.SerializeCompressed())
-
-	// Based on the disk representation of the node announcement generated
-	// above, we'll generate a node announcement that can go out on the
-	// network so we can properly sign it.
-	nodeAnn, err := selfNode.NodeAnnouncement(false)
-	if err != nil {
-		return nil, fmt.Errorf("unable to gen self node ann: %w", err)
-	}
-
-	// With the announcement generated, we'll sign it to properly
-	// authenticate the message on the network.
-	authSig, err := netann.SignAnnouncement(
-		s.nodeSigner, nodeKeyDesc.KeyLocator, nodeAnn,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to generate signature for "+
-			"self node announcement: %v", err)
-	}
-	selfNode.AuthSigBytes = authSig.Serialize()
-	nodeAnn.Signature, err = lnwire.NewSigFromECDSARawSignature(
-		selfNode.AuthSigBytes,
+	// We now get the self node which represents our node on the channel
+	// graph.
+	selfNode, err := s.getSelfNode(
```

Comment:
`get*` implies that this purely does a read. I suggest rather naming it something like `generateSelfNode`/`buildSelfNode`/`setSelfNode` to indicate that it is doing some processing given its arguments and also some persistence.